ament_cmake_test/ament_cmake_test/__init__.py

```diff
@@ -207,7 +207,7 @@ def log(msg, **kwargs):
                 break
             for i, encoding in enumerate(encodings):
                 try:
-                    decoded_line = line.decode(encoding)
+                    decoded_line = line.decode(encoding, errors='surrogateescape')
```

Comment:
@ejalaa12 IIUC, this breaks the fallback logic here.

We want to try all encodings before resorting to non-strict decoding. We can carry out such relaxed decode on line 213 instead.

Reply:
Indeed this is true, `errors='replace'` will always work for any encoding, so this would break the for-loop on the first iteration, rendering the whole for-loop unnecessary.

Instead we can fallback to `errors='replace'` instead of raising the `UnicodeDecodeError` exception. Btw, should this be silent ?

Reply:
> Btw, should this be silent ?

Good question. An status message would be good (a warning is too verbose for something downstream users cannot change).